### PR TITLE
fix(query engine) Change logging to use a dispatch

### PR DESCRIPTION
**Pull request by garrensmith**
The node api logging is still locking up when logging is enabled.
This is due to the fact we create a new registry for each new
query engine callback. This wraps the logging in a Dispatch so that
we don't need to create a registry for each engine call.

### CODE REVIEW

File: query-engine/query-engine-node-api/src/engine.rs [hunk 1]
```diff
@@ -173,29 +169,31 @@ impl QueryEngine {
 
         Ok(Self {
             inner: Arc::new(RwLock::new(Inner::Builder(builder))),
-            log_level,
-            log_queries,
-            log_callback,
+            logger: Self::create_log_dispatch(log_queries, log_level, log_callback),
```
Comment (pimeys) [inline query-engine/query-engine-node-api/src/engine.rs:172]: Maybe `logger::create_log_dispatch(..)`?

**Review by pimeys — APPROVED** [review]

File: query-engine/query-engine-node-api/src/engine.rs (hunk 1 above)
Comment (garrensmith) [reply query-engine/query-engine-node-api/src/engine.rs]: Good idea. I've changed it to that.

**Comment by garrensmith** [comment]
Logging passes client tests https://github.com/prisma/prisma/pull/12257

**Review by cprieto — COMMENTED** [review]
Nice work!

**Pull request merged**